ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -28,6 +53,75 @@ const OPERAND_IDX_MASK: u16 = 0x7fff;
 // The largest operand index we can express in 15 bits.
 const MAX_OPERAND_IDX: u16 = (1 << 15) - 1;
 
+/// [Instruction] to [InstrIdx] mapping used for stringifying instructions.
+///
+/// See [JitIRDisplay] for why this is required.
+pub(crate) struct LocalNumbers<'a> {
+    m: &'a Module,
+}
+
+impl<'a> LocalNumbers<'a> {
+    fn new(m: &'a Module) -> Result<LocalNumbers<'a>, CompilationError> {
+        Ok(Self { m })
+    }
+
+    /// Returns the [InstrIdx] of the specified [Instruction] in the [Module].
+    ///
+    /// # Panics
+    ///
+    /// Panics if the instruction isn't present in the module.
+    fn idx(&self, instr: &Instruction) -> Result<InstrIdx, CompilationError> {
+        // FIXME: This is inefficient.
+        for (idx, candidate) in self.m.instrs().iter().enumerate() {
+            if ptr::addr_eq(instr, candidate) {
+                return Ok(InstrIdx::new(idx)?);
+            }
+        }
+        panic!(); // Not found!
+    }
+}
+
+/// A trait for stringifying JIT IR.
+///
+/// We use this instead of [std::fmt::Display] because many IR constructs can't print something
+/// human-readable without querying the [Module] for more information. For example for to convert
```

Comment:
"for to"

Reply:
fixed in aa044fa7024bc9831c8a804719d2ef054c41d70a